### ISSUE TITLE
[3.5.rc6] Broken touch input on iOS

### ISSUE THREAD

**Issue by oeleo1** [godotengine/godot#62833]
### Godot version

3.5 RC6

### System information

iOS

### Issue description

Touch input on iOS became unresponsive in RC6 vs. RC5. I know about the change in the default for accumulated input.

But:
1. It works on Android
2. It doesn't work any longer on iOS (both with iOS touch delay = 0.01 in our project and the default of 0.15)
3. It still doesn't work on iOS by calling `Input.set_use_accumualted_input(true)`

I'll have to extract the touch controller code (again) to provide an MRP. Let me know if whoever tries RC6 with iOS cannot reproduce the problem. Thanks.

### Steps to reproduce

Try an existing project known to be working with touch input on iOS + Godot 3.5 RC6 :-)

### Minimal reproduction project

TBD

**Comment by akien-mga** [comment]
> It still doesn't work on iOS by calling `Input.set_use_accumualted_input(true)`

You shoudl call `Input.set_use_accumulated_input(false)` to turn it *off*, not *on*. It made turned *on* by default in RC 6.

**Comment by oeleo1** [comment]
> You should call Input.set_use_accumulated_input(false) to turn it off, not on. It was turned on by default in RC 6.

You are right, my bad.

> Could you confirm if turning it off solves the issue?

Unfortunately I can't. I need to do more investigation. Actually the situation is quite puzzling:
1. Calling `Input.set_use_accumulated_input(false)` doesn't change anything. `true` and `false` behave the same. Touch is dead. This is with a release build export.
2. Calling `Input.set_use_accumulated_input(false)` crashes the game if I export a debug build. Now, we have stout/stderr disabled on export and I know some debug printing has been changed there so I am really not sure what's going on.

**Comment by akien-mga** [comment]
Is it touch input on specific node types? Or touch input in general doesn't register any event ever?

**Comment by oeleo1** [comment]
> See https://github.com/godotengine/godot/commit/f64d5f40950505296f63a77d0d00bfe37f239695. This warning should only be enabled if you have the HDR property enabled on the Viewport

Thanks. In the project settings I have Rendering Quality HDR unchecked, and nowhere else can I see a HDR setting to disable, but that's off topic here.

 I'll come back on the touch issue with more info and hopefully an MRP

**Comment by oeleo1** [comment]
Adding the MRP - it's a nice simplistic gesture controller. In RC6 no events seem to be reaching the `_input(event)` handler of the `TouchScreenButton` node. There is no output. In RC5 all is fine.
[iOS_Touch.zip](https://github.com/godotengine/godot/files/9074024/iOS_Touch.zip)
.

**Comment by RandomShaper** [comment]
While I don't get why `Input.set_use_accumulated_input(false)` doesn't do the trick in RC6, I believe iOS was missing a call to flush the accumulated input events when it was enabled. I've created #62842 to sort that out.

**Comment by oeleo1** [comment]
Thanks, this is good news. I have actually reproduced with the MRP what @akien-mga asked for:
1. RC6 with `Input.set_use_accumulated_input(false)` added in the MRP as 1st thing in _ready() works
2. RC5 with `Input.set_use_accumulated_input(true)` doesn't work, the same way as RC6 without 1.

Now, our big project from which the MRP is extracted from probably crashes because the added `Input.set_use_accumulated_input()` statement is done late in the initialization or for some other reason that I'd need to investigate.

**Comment by RandomShaper** [comment]
Thanks for the report. Things really look like the PR will fix the issue.

**Comment by oeleo1** [comment]
Thanks - does this mean that it will work on iOS out of the box like it does on Android without the additional input config statement?

**Comment by RandomShaper** [comment]
Yes.

**Comment by akien-mga** [comment]
@oeleo1 Could you test this iOS template made with @RandomShaper's fix?

https://downloads.tuxfamily.org/godotengine/testing/Godot_v3.5-rc6%2bfixiostouch_export_templates.tpz

It should be compatible with the 3.5-rc6 macOS editor build, you can install the .tpz on top of existing templates to replace the `iphone.zip` template.

**Comment by oeleo1** [comment]
Yes, it works out of the box, without additional configuration. Good job!
